core/util.py

```diff
@@ -13,10 +13,28 @@
 from core.db import db
 
 POSTCODE_AREA_REGEX = r"(^[A-z]{1,2})[0-9R][0-9A-z]?"
+POSTCODE_PREFIX_REGEX = r"^[A-z]{1,2}"
 
 resources = Path(__file__).parent / ".." / "tests" / "resources"
 
 
+def get_postcode_prefix_set(postcodes: list[str]) -> set[str]:
+    """
+    Transform a list of postcodes into set of distinct postcode prefixes.
+
+    :param postcodes: A list of strings representing a UK postcode.
+    :return: A set of distinct postcode prefixes.
+    """
+    # TODO: FMD-241: Do we need to handles regex failures here, if all postcodes have already passed validation?
```

Comment:
It throws an Attribute error when it hits `postcode_prefix = postcode_area_matches.group(0)` as `postcode_area_matches` is None at that point. At a function level I feel like it would be good to handle it, but in reality this function will only ever get passed a list of postcodes that have already passed validation and so have passed a much more stringent regex that checks they have the correct start (plus everything else), and the regex here is more just to grab the first one or two characters that are A-Z - not sure if adding a catch here is good for safety or over thinking it?